### ISSUE TITLE
Modal overflows viewport when PyTorch MLP config form is selected

### ISSUE THREAD

**Issue by Sip4818** [Sip4818/TrainGrid#90]
## Problem

When creating a new run or sweep, selecting **PyTorch MLP** as the model populates the config form with many fields (architecture, optimizer, learning rate, epochs, patience, etc.). The `Modal` component has no `maxHeight` constraint — `dialogStyle` in `frontend/src/components/ui/Modal.tsx` sets `overflow: "auto"` but never caps the height, so the dialog grows to fit all its content. With enough fields the modal extends past the bottom of the viewport, making the submit button unreachable.

**Current dialog styles (relevant lines):**
```ts
const dialogStyle: React.CSSProperties = {
  // ...
  minWidth: "320px",
  maxWidth: "480px",
  width: "100%",
  overflow: "auto",        // scrollbar appears but height is uncapped
  // no maxHeight ← root cause
};
```

**Steps to reproduce:**
1. Navigate to a runs list or sweeps page.
2. Click **New Run** (or **New Sweep**).
3. In the model dropdown, select **PyTorch MLP**.
4. Observe that the form grows taller than the viewport — the submit/cancel buttons are below the fold and cannot be scrolled to.

## Proposed Fix

Cap the modal dialog height and let the **body** scroll while keeping the header (title + close button) and footer (submit/cancel buttons) pinned:

1. Add `maxHeight: "85vh"` (or similar) to `dialogStyle` so the dialog never exceeds 85% of the viewport height.
2. Restructure the modal's internal layout into three zones:
   - **Header** (fixed) — title + close button, always visible at the top.
   - **Body** (scrollable) — the `children` content area, gets `overflow-y: auto` and `flex: 1` so it takes remaining space and scrolls independently.
   - **Footer** (fixed) — submit/cancel buttons, always visible at the bottom. This may require the modal to accept a `footer` prop, or the forms that use the modal can wrap their buttons in a sticky container.
3. Use `display: "flex"` + `flexDirection: "column"` on the dialog so the three zones stack correctly and the body absorbs all remaining height.

**Alternative (simpler, if restructuring is too invasive):** Just add `maxHeight: "85vh"` to `dialogStyle`. Since `overflow: "auto"` is already set, the entire dialog content (including buttons) becomes scrollable. This is a one-line fix that solves the immediate problem, though it means users must scroll past the form to reach the buttons — acceptable but not ideal.

## Affected Component

- `frontend/src/components/ui/Modal.tsx` — the generic `Modal` component (fix here benefits all modals, not just the create-run form)

## Acceptance Criteria

- [x] When PyTorch MLP is selected in the create-run or create-sweep modal, the modal does not exceed the viewport height.
- [ ] The submit and cancel buttons are always reachable (either pinned at the bottom or reachable via scrolling within the modal).
- [ ] The fix works for all modals, not just the create-run form (generic `Modal` component change).
- [ ] Existing modal-related unit tests pass and `./check.sh` is green.
- [ ] Visually verified on a standard viewport (1080p) and a smaller viewport (e.g. laptop 768px height).

## Notes

- This should be done after PR #72 is merged to avoid conflicts.
- This is a UI-only fix — no backend changes required.
